tools/empath-split.py

```diff
@@ -252,31 +309,40 @@ def main():
     print_sources(sourcemap)
     return
 
-  paths = utils.read_file(args.paths_file).splitlines()
-  paths = [utils.normalize_path(path.strip()) for path in paths if path.strip()]
-  # To make /a/b/c and /a/b/c/ equivalent
-  paths = [path.rstrip(os.sep) for path in paths]
-  # Remove duplicates
-  paths = list(dict.fromkeys(paths))
+  content = utils.read_file(args.paths_file)
+  module_to_paths = parse_paths_file(content)
 
   # Compute {path: list of functions} map
-  path_to_funcs = get_path_to_functions_map(args.wasm, sourcemap, paths)
+  all_paths = []
+  for paths in module_to_paths.values():
+    all_paths.extend(paths)
+  path_to_funcs = get_path_to_functions_map(args.wasm, sourcemap, all_paths)
 
   # Write .manifest file
-  with tempfile.NamedTemporaryFile(suffix=".manifest", mode='w+', delete=args.preserve_manifest) as f:
+  with tempfile.NamedTemporaryFile(suffix=".manifest", mode='w+', delete=not args.preserve_manifest) as f:
     manifest = f.name
-    for i, path in enumerate(paths):
-      f.write(f'{i}\n')
-      if not path_to_funcs[path]:
-        diagnostics.warn(f'{path} does not match any functions')
+    for i, (module, paths) in enumerate(module_to_paths.items()):
+      funcs = set()
+      for path in paths:
+        if not path_to_funcs[path]:
+          diagnostics.warn(f'{path} does not match any functions')
+        funcs.update(path_to_funcs[path])
+      if not funcs:
+        diagnostics.warn(f"Module '{module}' does not match any functions")
+
       if args.verbose:
-        print(f'{path}: {len(path_to_funcs[path])} functions')
-        for func in path_to_funcs[path]:
-          print('  ' + func)
+        print(f'{module}: {len(funcs)} functions')
+        for path in paths:
+          if path in path_to_funcs:
+            print(f'  {path}: {len(path_to_funcs[path])} functions')
+          for func in path_to_funcs[path]:
+            print('    ' + func)
         print()
-      for func in path_to_funcs[path]:
+
+      f.write(f'{module}\n')
+      for func in funcs:
```

Comment:
Annoyingly If you want this to be deterministic then I think you need to make this a dict rather than set.

The `dict` in python became deterministic in 3.7 but the `set` remains unordered.    Annoying..